Add docs for execute, prove and verify, update docs (#2036)
software-mansion/scarb: website/docs/guides/creating-executable-package.md

<script setup>
import { data as rel } from "../../github.data";
import {data as constants} from "../../constants.data";
</script>

## Defining an Executable Package

> Start a new Scarb project with `Scarb new <project_name>`, and add the following to your `Scarb.toml` file:
> 
> 1. Specify that this package should compile to a Cairo executable by adding `[[target.executable]]` to your toml file (note that `lib` or `starknet-contract` targets cannot be executed in this way)
> 2. Add the `cairo_execute="{{ rel.stable.starknetPackageVersionReq }}"` plugin to your dependencies
> 3. Disable gas usage (gas is only supported for `lib` or `starknet-contract` targets) by adding `enable-gas = false` under the `[cairo]` section in your toml.

Reviewer: minor adjustments:
```suggestion
Start a new Scarb project with `scarb new <project_name>`. 
In your `Scarb.toml` file:

1. Set the package to compile to a Cairo executable by adding `[[target.executable]]` (note that `lib` or `starknet-contract` targets cannot be executed in this way).
2. Add the `cairo_execute="{{ rel.stable.starknetPackageVersionReq }}"` plugin to your dependencies.
3. Disable gas usage by adding `enable-gas = false` under the `[cairo]` section (gas is only supported for `lib` or `starknet-contract` targets).
```

Below we have an example of the manifest file of a simple executable

```
[package]
name = "test_execute"
version = "0.1.0"
edition = "{{ constants.edition }}"

[[target.executable]]

[cairo]
enable-gas = false

[dependencies]
cairo_execute = "{{ rel.stable.starknetPackageVersionReq }}"
```

Now we can move on to the code itself. An executable project must have **exactly one function** annotated with the `#[executable]` attribute. Consider the following simple `lib.cairo` file of an executable project:

```
#[executable]
fn main(num: u8) -> u8 {
    num
}
```

You can now run:

```
scarb execute -p test_execute --print-program-output --arguments 5
```

Where `test_execute` is the name of the package with the executable target (as defined in our Scarb.toml manifest)

> The above command runs our executable function within the `test-execute` package and prints the program's output segment, which contains a success bit (0 for success) followed by the Cairo Serde of main’s output or the panic reason in case of a panic.

Reviewer: 1. I suppose we should also add info that execution output is saved.
2. Maybe we should link smth like "for more information on `scarb execute`" here?
3. Not sure info about panic should be there, and not in `scarb execute` docs.

Author: > Not sure info about panic should be there, and not in scarb execute docs.

It's only a guide, not reference, so I think it can combine tips about both 🤔

Reviewer: What I rather meant is that it seems like info about panic is more essential to docs of `scarb execute` itself, rather than merely a guide, that is providing brief overview.

Author: Changed
